### PR TITLE
drivers: VCNL36825T: add trigger capability

### PR DESCRIPTION
**Pull request by deveritec-jusc**
Adds trigger capability to the Vishay VCNL36825T sensor.

It supports all 3 trigger modes, as well as the smart persistence feature which (according to the datasheet) improves the trigger speed.

### CODE REVIEW

File: drivers/sensor/vcnl36825t/vcnl36825t_trigger.c [hunk 1]
```diff
@@ -0,0 +1,321 @@
...
+		return;
+	}
+
+	rc = vcnl36825t_read(&config->i2c, VCNL36825T_REG_INT_FLAG, &reg_value);
+	if (rc < 0) {
+		LOG_ERR("error reading interrupt flag register %d", rc);
+		return;
+	}
+
+#if CONFIG_LOG
```
Comment (MaureenHelm) [inline drivers/sensor/vcnl36825t/vcnl36825t_trigger.c:115]: This shouldn't be needed

**Review by MaureenHelm — CHANGES_REQUESTED** [review]
Add this sensor to `tests/drivers/build_all/sensor/sensors_trigger_*.conf`

Also needs a rebase

File: drivers/sensor/vcnl36825t/vcnl36825t_trigger.c (hunk 1 above)
Comment (deveritec-rosc) [reply drivers/sensor/vcnl36825t/vcnl36825t_trigger.c:115]: fixed

**Review by MaureenHelm — DISMISSED** [review]

**Comment by deveritec-rosc** [comment]
Hm, it's now marked as stale. Any chance this will be part of Zephyr 3.8?

**Comment by deveritec-rosc** [comment]
Since it's now September, any chance this will be handled somewhen?

**Review by MaureenHelm — DISMISSED** [review]

**Review by ubieda — DISMISSED** [review]

**Comment by carlescufi** [comment]
@deveritec-rosc please rebase to resolve conflict

**Review by ubieda — DISMISSED** [review]

**Review by ubieda — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**